src/vos/lru_array.h

```diff
@@ -123,14 +123,19 @@ lrua_remove_entry(struct lru_sub *sub, uint32_t *head, struct lru_entry *entry,
 	/** Last entry in the list */
 	if (prev == entry) {
 		*head = LRU_NO_IDX;
-		return;
+	} else {
+		prev->le_next_idx = entry->le_next_idx;
+		next->le_prev_idx = entry->le_prev_idx;
+		if (idx == *head)
+			*head = entry->le_next_idx;
 	}
 
-	prev->le_next_idx = entry->le_next_idx;
-	next->le_prev_idx = entry->le_prev_idx;
-
-	if (idx == *head)
-		*head = entry->le_next_idx;
+	/*
+	 * If no free entries in the sub, then remove it from array free list (array->la_free_sub)
+	 * to avoid being searched when try to find free entry next time.
+	 */
+	if (head == &sub->ls_free && *head == LRU_NO_IDX && array->la_flags & LRU_FLAG_EVICT_MANUAL)
+		d_list_del_init(&sub->ls_link);
```

Comment:
why not just d_list_del ?

Reply:
For the purpose of checking 'd_list_empty' when re-add back to the free list.